dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -135,29 +137,33 @@
     private static final String TEI_LASTUPDATED = " tei.lastUpdated";
 
     private static final String GT_EQUAL = " >= ";
-
+    
     // -------------------------------------------------------------------------
     // Dependencies
     // -------------------------------------------------------------------------
 
     private final OrganisationUnitStore organisationUnitStore;
 
     private final StatementBuilder statementBuilder;
+    
+    private final Integer trackedEntityHardLimit;
 
     public HibernateTrackedEntityInstanceStore( SessionFactory sessionFactory, JdbcTemplate jdbcTemplate,
         ApplicationEventPublisher publisher, CurrentUserService currentUserService,
-        AclService aclService, OrganisationUnitStore organisationUnitStore, StatementBuilder statementBuilder )
+        AclService aclService, OrganisationUnitStore organisationUnitStore, StatementBuilder statementBuilder, DhisConfigurationProvider dhisConfigurationProvider )
     {
         super( sessionFactory, jdbcTemplate, publisher, TrackedEntityInstance.class, currentUserService, aclService,
             false );
 
         checkNotNull( statementBuilder );
         checkNotNull( organisationUnitStore );
+        checkNotNull( dhisConfigurationProvider );
 
         this.statementBuilder = statementBuilder;
         this.organisationUnitStore = organisationUnitStore;
+        this.trackedEntityHardLimit = Integer.valueOf(dhisConfigurationProvider.getProperty(ConfigurationKey.TRACKER_TRACKED_ENTITY_QUERY_LIMIT));
```

Comment:
Suggest moving the retrieval of the config value out of the constructor and into the method which uses it. Retrieving a config value is very fast/almost for free performance wise.

Reply:
Thanks for the feedback (and for the link).  I have updated the pull request after making the suggested changes and running the formatter.